client/components/notice/index.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+
+import React, { useEffect, FunctionComponent, useRef } from 'react';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'calypso/components/gridicon';
+
+/**
+ * Internal dependencies
+ */
+import { TimerHandle } from 'calypso/types';
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Module constants
+ */
+const GRIDICONS_WITH_DROP = [
+	'add',
+	'cross-circle',
+	'ellipsis-circle',
+	'help',
+	'info',
+	'notice',
+	'pause',
+	'play',
+	'spam',
+];
+
+/**
+ * Constants
+ */
+enum Status {
+	Error = 'is-error',
+	Info = 'is-info',
+	Success = 'is-success',
+	Warning = 'is-warning',
+	Plain = 'is-plain',
+}
+
+interface Props {
+	className?: string;
+	duration?: number;
```

Comment:
I think `duration` can also be `null`. This is how we declared it in https://github.com/Automattic/wp-calypso/blob/trunk/client/state/notices/types.ts#L29 as well. Is it possible we're not handling it properly here or there?

Reply:
I have searched all over Calypso and couldn't find a reference with `null` duration b05d68e

Reply:
Might be a remnant of the pre-Redux era. If we never provide a `null` for duration it would be nice to update the [notices  readme](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/notices/README.md#L43) and their [types](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/notices/types.ts#L29). Ideally, we should reuse as much as possible from there, too.